sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2747,6 +2747,17 @@ class Dataset[T] private[sql](
     }
   }
 
+  /**
+   * Collect a Dataset as ArrowPayload byte arrays and serve to PySpark.
+   */
+  private[sql] def collectAsArrowToPython(): Int = {
+    val payloadRdd = toArrowPayloadBytes()
+    val payloadByteArrays = payloadRdd.collect()
```

Comment:
Right, but since it had a small effect on performance I thought it would be better to avoid for now.  I have also been experimenting working with the Arrow `StreamWriter` which does reduce memory usage and improves performance, but required some additional changes to Spark so I think it would be better to propose as a follow up PR.